### PR TITLE
Process of adding Whaley's Locations (1 of 9)

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
**THIS PR WAS REMADE [HERE](https://github.com/CleverRaven/Cataclysm-DDA/pull/25070)**

_This process intends to add Whaley's Location Mods to the game (those that haven't already been added). Before getting to adding said locations I thought it would be a good idea to combine some mapgen files and organize the mapgen folder a bit._

---

This **first** step combines many mapgen files (ex. combines pool.json & pool1.json into just pool.json). It also moves and renames several files to organize the mapgen folder. But, because this PR was getting fairly lengthy I'm splitting this organizational stage into 2 steps. 

**This PR is NOT dependent on any other step.**

Discussion for the addition of Whaley's Mods can be found **[here](https://github.com/CleverRaven/Cataclysm-DDA/issues/24932)**.

This step also contains a fix for issue #24825.

---

**[2 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25072)** - Finish combining/renaming/moving mapgen files (specifically Whaley's early additions in the "parks_recreation" and "small_town_buildings" folders).

**[3 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24924)** - Move Whaley's official (in-game) mod, "More Buildings" to the vanilla game.

**[4 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24989)** - Adds Whaley's Zoo (from his "Big Buildings" mod).

**[5 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24990)** - Adds Whaley's Stadium (from his "Big Buildings mod).

**[6 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24992)** - Adds the remainder of Whaley's "Big Buildings" mod.

**[7 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/24998)** - Adds Whaley's "City Entertainment" mod.

**[8 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25000)** - Adds Whaley's "Seedy Spots" mod.

**[9 of 9](https://github.com/CleverRaven/Cataclysm-DDA/pull/25002)** - Adds Whaley's "River Buildings" mod.

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
What needs to be here exactly? @ZhilkinSerg

**Comment by ZhilkinSerg** [comment]
Coffee trees were added to coffee shop and botanical gardens in #24829, so you need to update these files accordingly.

**Comment by cosmiccoincidence** [comment]
- Updated with the coffee trees at coffee shop and botanical garden.

And while I was here I decided to do a fix for #24825 (removing marloss flowers from botanical and public gardens) which resulted in my changing a few things around. 

- Removed marloss and fungal flowers from the greenhouse, moved the strawberry and blueberry bushes into the greenhouse, and moved the coffee tree to the old strawberry and blueberry spot in the middle (just north of the player in my screenshot).

- Also added a 1/3 chance for the plants in the greenhouse to grow on any of their plot tiles to add randomness and fullness.

https://i.imgur.com/Xb30L3B.png

File: data/json/mapgen/abandoned_barn.json [hunk 1]
```diff
@@ -89,7 +89,7 @@
         "|": "t_wall_wood"
       }
     },
-    "om_terrain": "desolatebarn",
```
Comment (ZhilkinSerg) [inline data/json/mapgen/abandoned_barn.json]: Changing id of overmap terrains will cause issues when loading saved game until you provide a migration code (see https://github.com/CleverRaven/Cataclysm-DDA/blob/b8ef478f8998929fc3ab53c9a24ccbbece595cb3/src/savegame.cpp#L317-L711).

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild

**Comment by ZhilkinSerg** [comment]
A couple of files are not linted:

image

image

Please use http://dev.narc.ro/cataclysm/format.html to lint files.

**Comment by ZhilkinSerg** [comment]
These files still needs conflict resolution:

data/json/mapgen/coffee_shop2.json
data/json/mapgen/cosmics_additions/GardenBotanical.json

File: .gitignore [hunk 2]
```diff
@@ -156,3 +156,4 @@ stage
 
 # Apple Xcode IDE
 Xcode/
+data/json/mapgen/garden_botanical.json
```
Comment (ZhilkinSerg) [inline .gitignore]: Should not be ignored.

Comment (cosmiccoincidence) [reply .gitignore]: fixed

**Comment by cosmiccoincidence** [comment]
Went to merge master into this branch to make sure of no conflicting files but it made too many changes to my changes (re-adds files I deleted) so I reverted it.

**Comment by cainiaowu** [comment]
Wow, now we are going back though time now. 

Check the changed files.

**Comment by Qrox** [comment]
> Wow, now we are going back though time now

I noticed that too. Can we revert this merge and let @cosmiccoincidence do the proper conflict resolution?

**Comment by cosmiccoincidence** [comment]
The commit you commented on was a revert to a previous commit that WAS fucked up. Hence why I reverted it.

**Comment by Qrox** [comment]
Reverting a merge commit will revert all changes made at the remote, not changes made to the local branch, so you actually reverted all changes made to the master branch since your second-last merge.

**Comment by cosmiccoincidence** [comment]
False.

I merged master into my branch not my branch into the master. So reverted the commit reverted me merging master TO my branch. It reverted nothing in the master.

**Comment by Qrox** [comment]
Ok, it turns out that reverting a merge is more complicated than I thought, but you can check the changes made to files such as `astyle_blacklist` by your revert commit: lines changed in master branch at commit 7f8a9ce and 4a153b1 are reverted by it.

**Comment by cosmiccoincidence** [comment]
And I don't mean to sound rude if I did (it's late here and i'm close to going to bed) so sorry if I came across that way... And I appreciate you taking time to look at this.

It actually does appear that now that this is merged it DID affect the master... Instead of deleting a change, reverting it just does the opposite... and with the merge of the PR, it reverted the changes in the master now.

I guess as a result I must merge master again to my branch to have those same changes that were already merged (by other PRs) to stay consistent with the master..

**Comment by cainiaowu** [comment]
A revert in git is actually making a whole new commit that doing the opposite thing. https://git-scm.com/docs/git-revert

> Note: git revert is used to record some new commits to reverse the effect of some earlier commits (often only a faulty one). 

You should use "git rebase" to rebase your change other than revert function.
@Rivet-the-Zombie check this please.

**Comment by cosmiccoincidence** [comment]
I've been using github desktop, I'm not aware of a 'git rebase'.

**Comment by cosmiccoincidence** [comment]
Can I just revert my revert, adding said changes back in (albeit into this PR)?

**Comment by cainiaowu** [comment]
Well, you can, but the change will only stay on your branch. 

It needs someone with master merge right to fix the master.

Also use a better git gui tool like sourcetree, github desktop is just meh (me personal bias).

**Pull request merged**